mart/attack/perturber.py

```diff
@@ -70,6 +72,15 @@ def create_from_tensor(tensor):
             else:
                 raise NotImplementedError
 
+        # Select input according to index
+        index = []
+        for i in range(max(len(self.shape), len(input.shape))):
+            input_shape = input.shape[i]
+            desired_shape = self.shape[i]
+
+            index.append(slice(0, desired_shape or input_shape))
+        input = input[index]
```

Comment:
This is a terrible implementation but fulfill the spirit of what we need to do to make MART work.